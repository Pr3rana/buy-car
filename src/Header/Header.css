header{
    height: 80px;
    display: flex;
    justify-content: flex-end;
    align-items: center;
    background-color: #ededed;
}

.fill-remaining-space {
  flex: auto;
}

.logo{
  display: inline-block;
  height: 40px;
  width: 200px;
  padding: 24px;
}

ul {
    list-style-type: none;
    display: contents;
    margin: 0;
    padding: 0;
    overflow: hidden;
    height: 100%;
  }
  
  li {
    display: inline-block;
    text-align: center;
  }
  
  li a {
    display: inline-block;
    font-size: 18px;
    display: inline;
    color: #4a4a4a;
    text-align: center;
    padding: 24px;
    text-decoration: none;
  }
  
  li a:hover:not(.active) {
    background-color: #ea7f28;
    color: #ededed;
  }
  
  /* .active {
    background-color: #4CAF50;
  } */
