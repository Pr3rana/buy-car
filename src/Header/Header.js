import './Header.css';
import logo from '../logo.png';

const Header = () => {
    return ( 
        <header>
            <img alt="logo" className="logo" src={logo}/>
            <div className="fill-remaining-space"></div>
            <ul>
                <li><a href="javascript:void(0)">Purchase</a></li>
                <li><a href="javascript:void(0)">My Orders</a></li>
                <li><a href="javascript:void(0)">Sells</a></li>
            </ul>
        </header>
     );
}
 
export default Header;
