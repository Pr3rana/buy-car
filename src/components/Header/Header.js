import './Header.css';
import logo from '../../logo.png';

const Header = ({headerNavList=[], brandLogo=logo}) => {
    return ( 
        <header data-testid="header">
            <img data-testid ="brand-logo" alt="brand-logo" className="logo" src={brandLogo}/>
            <div className="fill-remaining-space"></div>
            {headerNavList.length>0 && <ul data-testid ="header-list-container">
               {
                   headerNavList.map((nav,index)=>(
                    <li data-testid ="header-list-item" key={index}><a href="/" onClick={(e)=>e.preventDefault()}>{nav}</a></li>
                   ))
               }
            </ul>}
        </header>
     );
}
 
export default Header;
