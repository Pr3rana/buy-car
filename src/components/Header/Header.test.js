import { render, screen, cleanup } from '@testing-library/react';
import Header from './Header';
import logo from '../../logo.png';

afterEach(cleanup);

describe("Header render", ()=>{
    test('Initial Render', () => {
        const { container } = render(<Header/>)
        expect(container.firstChild).toMatchSnapshot(`
            <header data-testid="header">
            <img data-testid ="brand-logo" alt="brand-logo" className="logo" src={requiredLogo}/>
            <div className="fill-remaining-space"></div>
            </header>
        `)
        expect(screen.queryByTestId('header-list-container')).not.toBeInTheDocument()
    });
});

describe("Header props",()=>{
    test("Re-render the Header with different props",()=>{
        const { container } = render(<Header headerNavList = {["test", "test"]} requiredLogo={logo} />)
        expect(screen.getByTestId('header')).toHaveTextContent('test');
        expect(screen.queryAllByText('test')).toHaveLength(2);
        expect(screen.queryByTestId('header-list-container')).toBeInTheDocument()
        expect(screen.getByTestId('header-list-container').firstChild).toHaveTextContent('test');
        expect(container.firstChild).toMatchSnapshot(`
            <header data-testid="header">
                <img data-testid ="brand-logo" alt="brand-logo" className="logo" src={requiredLogo}/>
                <div className="fill-remaining-space"></div>
                <ul data-testid ="header-list-container">
                    <li data-testid ="header-list-item" key=0><a href="/" onClick={(e)=>e.preventDefault()}>test</a></li>
                    <li data-testid ="header-list-item" key=1><a href="/" onClick={(e)=>e.preventDefault()}>test</a></li>
                </ul>
            </header>
        `)
    })
})