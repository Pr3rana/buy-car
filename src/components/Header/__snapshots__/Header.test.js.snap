// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header props Re-render the Header with different props: 
            <header data-testid="header">
                <img data-testid ="brand-logo" alt="brand-logo" className="logo" src={requiredLogo}/>
                <div className="fill-remaining-space"></div>
                <ul data-testid ="header-list-container">
                    <li data-testid ="header-list-item" key=0><a href="/" onClick={(e)=>e.preventDefault()}>test</a></li>
                    <li data-testid ="header-list-item" key=1><a href="/" onClick={(e)=>e.preventDefault()}>test</a></li>
                </ul>
            </header>
         1`] = `
<header
  data-testid="header"
>
  <img
    alt="brand-logo"
    class="logo"
    data-testid="brand-logo"
    src="logo.png"
  />
  <div
    class="fill-remaining-space"
  />
  <ul
    data-testid="header-list-container"
  >
    <li
      data-testid="header-list-item"
    >
      <a
        href="/"
      >
        test
      </a>
    </li>
    <li
      data-testid="header-list-item"
    >
      <a
        href="/"
      >
        test
      </a>
    </li>
  </ul>
</header>
`;

exports[`Header render Initial Render: 
            <header data-testid="header">
            <img data-testid ="brand-logo" alt="brand-logo" className="logo" src={requiredLogo}/>
            <div className="fill-remaining-space"></div>
            </header>
         1`] = `
<header
  data-testid="header"
>
  <img
    alt="brand-logo"
    class="logo"
    data-testid="brand-logo"
    src="logo.png"
  />
  <div
    class="fill-remaining-space"
  />
</header>
`;
